drake/systems/plants/xmlUtil.cpp

```diff
@@ -0,0 +1,204 @@
+{
+  const char* attr = node->Attribute(attribute_name);
+  if (attr) {
+    std::stringstream s(attr);
+    s >> val(0) >> val(1) >> val(2) >> val(3);
+    return true;
+  }
+  return false;
+}
+
+bool parseVectorValue(tinyxml2::XMLElement* node, const char* element_name, Eigen::Vector3d &val)
+{
+  XMLElement* elnode = node->FirstChildElement(element_name);
+  if (elnode && elnode->FirstChild()) {
+    std::stringstream s(elnode->FirstChild()->Value());
+    s >> val(0) >> val(1) >> val(2);
+    return true;
+  }
+  return false;
+}
+
+bool parseVectorValue(tinyxml2::XMLElement* node, const char* element_name, Eigen::Vector4d &val)
+{
+  XMLElement* elnode = node->FirstChildElement(element_name);
+  if (elnode && elnode->FirstChild()) {
+    std::stringstream s(elnode->FirstChild()->Value());
+    s >> val(0) >> val(1) >> val(2) >> val(3);
+    return true;
+  }
+  return false;
+}
+
+bool parseStringValue(tinyxml2::XMLElement* node, const char* element_name, std::string &val)
+{
+  XMLElement* elnode = node->FirstChildElement(element_name);
+  if (elnode && elnode->FirstChild()) {
+    val = elnode->FirstChild()->Value();
+    return true;
+  }
+  return false;
+}
+
+
+void originAttributesToTransform(tinyxml2::XMLElement *node, Eigen::Isometry3d &T)
+{
+  Eigen::Vector3d rpy=Eigen::Vector3d::Zero(), xyz=Eigen::Vector3d::Zero();
+
+  parseVectorAttribute(node,"xyz",xyz);
+  parseVectorAttribute(node,"rpy",rpy);
+
+  T.matrix() << rpy2rotmat(rpy), xyz, 0,0,0,1;
+}
+
+void poseValueToTransform(tinyxml2::XMLElement *node, const PoseMap& pose_map, Eigen::Isometry3d &T, const Eigen::Isometry3d& T_default_frame)
+{
+  Eigen::Vector3d rpy=Eigen::Vector3d::Zero(), xyz=Eigen::Vector3d::Zero();
+  const char* strval = node->FirstChild()->Value();
+  if (strval) {
+    std::stringstream s(strval);
+    s >> xyz(0) >> xyz(1) >> xyz(2) >> rpy(0) >> rpy(1) >> rpy(2);
+  }
+  T.matrix() << rpy2rotmat(rpy), xyz, 0,0,0,1;
+
+  const char* attr = node->Attribute("frame");
+  if (attr && strlen(attr)>0) {
+    std::string frame;
+    std::stringstream s(attr);
+    s >> frame;
+    assert(0 && "this has not been tested yet"); // and could cause problems with the default pose assumptions in the sdf parser, which I simply had to guess
+    Eigen::Isometry3d T_frame = pose_map.at(frame);  // will throw an exception if the frame is not found.  that is the desired behavior.
+    T = T_frame*T;
+  } else {
+    T = T_default_frame*T;
+  }
+}
+
+string exec(string cmd)
+{
+  FILE* pipe = POPEN(cmd.c_str(), "r");
+  if (!pipe) return "ERROR";
+  char buffer[128];
+  string result = "";
+  while(!feof(pipe)) {
+    if(fgets(buffer, 128, pipe) != NULL)
+      result += buffer;
+  }
+  PCLOSE(pipe);
+  return result;
+}
+
+void searchDirectory(map<string,string> &package_map, string path)
+{
+  string token, t;
+  istringstream iss(path);
+
+  while (getline(iss,token,':')) {
+    istringstream p(exec("find -L "+token+" -iname package.xml"));
```

Comment:
no.  had already opened an issue about it:  
https://github.com/RobotLocomotion/drake/issues/1718

it's been in our workflow since Mar, 2015 (cc739b59618f5f7c35578e56c84ef4ae00513c4c); I only moved this to a different file.